agent/proto/resourcesmonitor/resources_monitor_test.go

```diff
@@ -0,0 +1,235 @@
+					{
+						Enabled: true,
+						Path:    "/",
+					},
+				},
+			},
+			datapointsPusher: func(_ context.Context, req *proto.PushResourcesMonitoringUsageRequest) (*proto.PushResourcesMonitoringUsageResponse, error) {
+				require.Len(t, req.Datapoints, 20)
+				require.Nil(t, req.Datapoints[0].Memory)
+				require.NotNil(t, req.Datapoints[0].Volumes)
+				require.Equal(t, &proto.PushResourcesMonitoringUsageRequest_Datapoint_VolumeUsage{
+					Volume: "/",
+					Total:  100000,
+					Used:   50000,
+				}, req.Datapoints[0].Volumes[0])
+
+				return &proto.PushResourcesMonitoringUsageResponse{}, nil
+			},
+			fetcher: &fetcher{
+				totalMemory: 0,
+				usedMemory:  0,
+				errMemory:   assert.AnError,
+				totalVolume: 100000,
+				usedVolume:  50000,
+			},
+			numTicks: 20,
+		},
+		{
+			// If one of the resources fails to be fetched, the datapoints still should be pushed with the other resources.
+			name: "ErrorFetchingVolume",
+			config: &proto.GetResourcesMonitoringConfigurationResponse{
+				Config: &proto.GetResourcesMonitoringConfigurationResponse_Config{
+					NumDatapoints:             20,
+					CollectionIntervalSeconds: 1,
+				},
+				Volumes: []*proto.GetResourcesMonitoringConfigurationResponse_Volume{
+					{
+						Enabled: true,
+						Path:    "/",
+					},
+				},
+			},
+			datapointsPusher: func(_ context.Context, req *proto.PushResourcesMonitoringUsageRequest) (*proto.PushResourcesMonitoringUsageResponse, error) {
+				require.Len(t, req.Datapoints, 20)
+				require.Len(t, req.Datapoints[0].Volumes, 0)
+
+				return &proto.PushResourcesMonitoringUsageResponse{}, nil
+			},
+			fetcher: &fetcher{
+				totalMemory: 16000,
+				usedMemory:  8000,
+				totalVolume: 0,
+				usedVolume:  0,
+				errVolume:   assert.AnError,
+			},
+			numTicks: 20,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			var (
+				logger       = slog.Make(sloghuman.Sink(os.Stdout))
+				clk          = quartz.NewMock(t)
+				counterCalls = 0
+			)
+
+			datapointsPusher := func(ctx context.Context, req *proto.PushResourcesMonitoringUsageRequest) (*proto.PushResourcesMonitoringUsageResponse, error) {
+				counterCalls++
+				return tt.datapointsPusher(ctx, req)
+			}
+
+			pusher := &datapointsPusherMock{
+				PushResourcesMonitoringUsageFunc: datapointsPusher,
+			}
+
+			monitor := resourcesmonitor.NewResourcesMonitor(logger, clk, tt.config, tt.fetcher, pusher)
+			require.NoError(t, monitor.Start(ctx))
+
+			for i := 0; i < tt.numTicks; i++ {
+				_, waiter := clk.AdvanceNext()
+				require.NoError(t, waiter.Wait(ctx))
+			}
+
+			// expectedCalls is computed with the following logic :
+			// We have one call per tick, once reached the ${config.NumDatapoints}.
+			expectedCalls := tt.numTicks - int(tt.config.Config.NumDatapoints) + 1
+			require.Equal(t, expectedCalls, counterCalls)
+			cancel()
```

Comment:
Nit: I don't think you need this here since you have a `defer cancel()` already which will be invoked after this.